presto-main/src/test/java/com/facebook/presto/operator/TestHashBuilderOperator.java

```diff
@@ -0,0 +1,217 @@
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestHashBuilderOperator
+{
+    private static final SingleStreamSpillerFactory SINGLE_STREAM_SPILLER_FACTORY = new DummySpillerFactory();
+
+    private ExecutorService executor;
+    private ScheduledExecutorService scheduledExecutor;
+
+    @BeforeMethod
+    public void setUp()
+    {
+        // Before/AfterMethod is chosen here because the executor needs to be shutdown
+        // after every single test case to terminate outstanding threads, if any.
+
+        // The line below is the same as newCachedThreadPool(daemonThreadsNamed(...)) except RejectionExecutionHandler.
+        // RejectionExecutionHandler is set to DiscardPolicy (instead of the default AbortPolicy) here.
+        // Otherwise, a large number of RejectedExecutionException will flood logging, resulting in Travis failure.
+        executor = new ThreadPoolExecutor(
+                0,
+                Integer.MAX_VALUE,
+                60L,
+                SECONDS,
+                new SynchronousQueue<Runnable>(),
+                daemonThreadsNamed("test-executor-%s"),
+                new ThreadPoolExecutor.DiscardPolicy());
+        scheduledExecutor = newScheduledThreadPool(2, daemonThreadsNamed("test-scheduledExecutor-%s"));
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void tearDown() throws Exception
+    {
+        executor.shutdownNow();
+        scheduledExecutor.shutdownNow();
+        executor.awaitTermination(365, DAYS);
+        scheduledExecutor.awaitTermination(365, DAYS);
+    }
+
+    @DataProvider(name = "hashBuildTestValues")
+    public static Object[][] hashJoinTestValuesProvider()
+    {
+        return new Object[][] {
+                {true, false},
+                {false, false},
+                {true, true},
+                {false, true},
+        };
+    }
+
+    @Test(dataProvider = "hashBuildTestValues")
+    public void testMemoryTracking(boolean buildHashEnabled, boolean spillEnabled)
+    {
+        // Build the resources you need
+        RowPagesBuilder buildPages = rowPagesBuilder(buildHashEnabled, Ints.asList(0), ImmutableList.of(VARCHAR, BIGINT, BIGINT));
+        Random r = new Random(384013974023187L);
+        for (int i = 0; i < 1000; i++) {
+            buildPages.addSequencePage(1024, r.nextInt(), r.nextInt(), r.nextInt());
+        }
+
+        HashBuilderOperator buildOperator = createTestOperator(buildPages, spillEnabled);
+
+        assertEquals(buildOperator.getState(), State.CONSUMING_INPUT);
```

Comment:
Static import `CONSUMING_INPUT`

Reply:
You mean, import the whole path, instead of just State? Good call